### PR TITLE
feat: Support Transit Gateway XAC connection actions

### LINKED ISSUE #4347: Support for accepting a transit gateway connection
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Scenario:
- Transit geteway used to connect VPCs located in different account
- When adding a connection to a VPC in a different account, there is a need for a user with sufficient priviledge in the target account to accept the connection. 

There is API support for accepting connection - see https://cloud.ibm.com/apidocs/transit-gateway#create-transit-gateway-connection-actions 

However, the provider is currently missing this capability.

### New or Affected Resource(s)

* ibm_tg_connection

### References

* #0000

### PR DESCRIPTION
**Pull request by sarah-ulmer**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #4347

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```
**Note:** The terraform acceptance test framework does not support multiple providers configs with differing api keys. We can not use the acceptance tests for this feature because it needs two accounts/apikeys.

#### Manual tests
Config:
[main.txt](https://github.com/IBM-Cloud/terraform-provider-ibm/files/10854318/main.txt)
[provider.txt](https://github.com/IBM-Cloud/terraform-provider-ibm/files/10854325/provider.txt)

Terraform apply
[terraform-create-xac-connection-and-approve.txt](https://github.com/IBM-Cloud/terraform-provider-ibm/files/10854775/terraform-create-xac-connection-and-approve.txt)
[terraform-delete-xac-connection.txt](https://github.com/IBM-Cloud/terraform-provider-ibm/files/10854835/terraform-delete-xac-connection.txt)

#### Feature Support
This feature introduces a new resource `ibm_tg_connection_action` to add support for managing cross-account connections. A user can create a transit gateway connection that connects to a resource in another account. The second account must approve or reject the connection by submitting the approve/reject action to the '/transit_gateways/{transit_gateway_id}/connections/{id}/actions' API. Guidance for cross account connections is documented here: https://cloud.ibm.com/docs/transit-gateway?topic=transit-gateway-adding-cross-account-connections&interface=api

### CODE REVIEW

File: ibm/provider/provider.go [hunk 1]
```diff
@@ -1058,6 +1058,7 @@ func Provider() *schema.Provider {
 			// //Added for Transit Gateway
 			"ibm_tg_gateway":                  transitgateway.ResourceIBMTransitGateway(),
 			"ibm_tg_connection":               transitgateway.ResourceIBMTransitGatewayConnection(),
+			"ibm_tg_connection_actions":       transitgateway.ResourceIBMTransitGatewayConnectionActions(),
```
Comment (kavya498) [inline ibm/provider/provider.go:1061]: Will this resource create multiple actions?

If not, can we name it to `ibm_tg_connection_action`?
and also change the name accordingly in all the references, filenames and in docs.

File: website/docs/r/tg_connection_actions.html.markdown [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+subcategory: "Transit Gateway"
+layout: "ibm"
+page_title: "IBM : tg_connection_actions"
+description: |-
+  Perform actions on a connection for a Transit Gateway
+---
+
+# ibm_tg_connection_action
```
Comment (kavya498) [inline website/docs/r/tg_connection_actions.html.markdown:9]: make sure this name is as same as resource name

page title says `tg_connection_actions` but this says `_action`

**Review by kavya498 — CHANGES_REQUESTED** [review]

File: ibm/provider/provider.go (hunk 1 above)
Comment (sarah-ulmer) [reply ibm/provider/provider.go]: I renamed it to 'ibm_tg_connection_action' and updated all references and docs.

File: website/docs/r/tg_connection_actions.html.markdown (hunk 2 above)
Comment (sarah-ulmer) [reply website/docs/r/tg_connection_actions.html.markdown:9]: I renamed the resource to '_action', so this is correct

**Comment by sarah-ulmer** [comment]
@kavya498 Thank you for your feedback, I've addressed the changes. Can you please re-review and merge?

I also ran the manual tests again to verify the changes work: [tf-create-xac-action-2.txt](https://github.com/IBM-Cloud/terraform-provider-ibm/files/10936822/tf-create-xac-action-2.txt)

**Pull request merged**